File: ios/Runner/ChannelInfo.swift
//
//  ChannelInfo.swift
//  BVISION
//
//  Created by patrick on 2018/5/28.
//  Copyright © 2018 wiatec. All rights reserved.
//

//
//  VodChannelInfo.swift
//  BVISION
//
//  Created by patrick on 2018/5/28.
//  Copyright © 2018 wiatec. All rights reserved.
//
import SwiftyJSON

struct ChannelInfo {
    var id: Int = 0
    var channelId: String = ""
    var tag: String = ""
    var label: String = ""
    var name: String = ""
    var type: Int = 0
    var visible: Bool = false
    var blocked: Bool = false
    var sync: Int = 0
    var flag: Int = 0
    var url: String = ""
    var icon: String = ""
    var remark: String = ""
    
    
    init(_ jsonData: JSON) {
        id = jsonData["id"].intValue
        channelId = jsonData["channelId"].stringValue
        tag = jsonData["tag"].stringValue
        label = jsonData["label"].stringValue
        name = jsonData["name"].stringValue
        type = jsonData["type"].intValue
        visible = jsonData["visible"].boolValue
        blocked = jsonData["blocked"].boolValue
        sync = jsonData["sync"].intValue
        flag = jsonData["flag"].intValue
        url = jsonData["url"].stringValue
        icon = jsonData["icon"].stringValue
        remark = jsonData["remark"].stringValue
    }
}

File: lib/entities/channel_info.dart
import 'package:json_annotation/json_annotation.dart';
part 'channel_info.g.dart';

@JsonSerializable()
class ChannelInfo extends Object{

  @JsonKey()
  int id;
  @JsonKey()
  String channelId;
  @JsonKey()
  String tag;
  @JsonKey()
  String label;
  @JsonKey()
  String name;
  @JsonKey()
  int type;
  @JsonKey()
  bool visible;
  @JsonKey()
  bool blocked;
  @JsonKey()
  int sync;
  @JsonKey()
  int flag;
  @JsonKey()
  String url;
  @JsonKey()
  String icon;
  @JsonKey()
  String remark;

  ChannelInfo(this.id, this.channelId, this.tag, this.label, this.name,
      this.type, this.visible, this.blocked, this.sync, this.flag, this.url,
      this.icon, this.remark);


  @override
  String toString() {
    return 'ChannelInfo{id: $id, channelId: $channelId, tag: $tag, label: $label, name: $name, type: $type, visible: $visible, blocked: $blocked, sync: $sync, flag: $flag, url: $url, icon: $icon, remark: $remark}';
  }

  factory ChannelInfo.fromJson(Map<String, dynamic> json) => _$ChannelInfoFromJson(json);



}